enzyme/Enzyme/Enzyme.cpp

```diff
@@ -130,14 +139,45 @@ class Enzyme : public ModulePass {
     auto FT = cast<Function>(fn)->getFunctionType();
     assert(fn);
 
-    if (EnzymePrint)
-      llvm::errs() << "prefn:\n" << *fn << "\n";
+    IRBuilder<> Builder(CI);
 
-    std::vector<DIFFE_TYPE> constants;
-    SmallVector<Value *, 2> args;
+    bool fnsret = false;
+    if (cast<Function>(fn)->hasParamAttribute(0, Attribute::StructRet)) {
+      Type *fnsrety = cast<PointerType>(FT->getParamType(0));
+      Type *cisrety = cast<PointerType>(CI->getArgOperand(0)->getType());
 
-    unsigned truei = 0;
-    IRBuilder<> Builder(CI);
+      if (fnsret == cisret) {
```

Comment:
I might remove fnsret here since its always false. I might in fact remove the variable entirely and instead do truei++ here.

Reply:
Fixed it. It was supposed to be `(fnsrety != cisrety)`. I also removed `fnsret`